pkg/apis/operator/v1/intrusiondetection_types.go

```diff
@@ -42,6 +42,7 @@ type IntrusionDetectionStatus struct {
 // of this resource is supported. It must be named "tigera-secure".
 // +k8s:openapi-gen=true
 // +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
```

Comment:
Is the `+genclient:nonNamespaced` still needed?

Reply:
The `genclient` stuff is all for building k8s golang clients.

The `kubebuilder` tags are for CRD yaml files. 

So, I think they are both used for different things.